src/index.js

```diff
@@ -86,6 +86,12 @@ function fetch (uri, opts) {
           reject(new FetchError(`redirect location header missing at: ${request.url}`, 'invalid-redirect'))
           return
         }
+        // Remove authorization if changing hostnames (but not if just
+        // changing ports or protocols).  This matches the behavior of request:
+        // https://github.com/request/request/blob/b12a6245/lib/redirect.js#L134-L138
+        if (url.parse(request.url).hostname !== url.parse(res.headers.location).hostname) {
```

Comment:
I think we need to emulate request's behavior a little more closely here: `res.headers.location` can be `/foo` or `foo.com/x`, which would not have a `hostname` property when put through `url.parse()`.

It might be nicer to have code that's more explicit about detecting these different cases, instead of using request's concise but not-very-explicit implementation of it.

This patch looks otherwise fine, and I'm mostly fine with the tests as they are, modulo adding tests for `Location: /foo` and `Location: foo.com` and such